include/zephyr/kernel/internal/sched_priq.h

```diff
@@ -10,6 +10,18 @@
 #include <zephyr/sys/dlist.h>
 #include <zephyr/sys/rb.h>
 
+#define K_HIGHEST_THREAD_PRIO (-CONFIG_NUM_COOP_PRIORITIES)
+#define K_LOWEST_THREAD_PRIO CONFIG_NUM_PREEMPT_PRIORITIES
+#define K_IDLE_PRIO K_LOWEST_THREAD_PRIO
+#define K_HIGHEST_APPLICATION_THREAD_PRIO (K_HIGHEST_THREAD_PRIO)
+#define K_LOWEST_APPLICATION_THREAD_PRIO (K_LOWEST_THREAD_PRIO - 1)
+#define K_THRAD_PRIO_NUM (K_LOWEST_THREAD_PRIO - K_HIGHEST_THREAD_PRIO + 1)
+#define PRI_QUEUE_BITMAP_SIZE ((K_THRAD_PRIO_NUM + BIT_NUM - 1) / BIT_NUM)
+#define PRI_BITMAP_ITERATOR PRI_QUEUE_BITMAP_SIZE
+#define BIT_NUM 32
```

Comment:
Generally better to compute this via sizeof() at the spot where it's used, e.g. `8 * sizeof(pq->bitmask[0]` or whatever.

Reply:
`#define BIT_NUM (8 * sizeof(((struct _priq_mq *)0)->bitmask[0]))` will cause `invalid use of undefined type 'struct _priq_mq'`